RealmWeaver/ModuleWeaver.cs

```diff
@@ -230,8 +230,18 @@ private void WeaveType(TypeDefinition type, Dictionary<string, Tuple<MethodRefer
             }
         }
 
+        var objectConstructor = type.GetConstructors()
+            .SingleOrDefault(c => c.Parameters.Count == 0 && c.IsPublic && !c.IsStatic);
+        if (objectConstructor == null)
+        {
+            var nonDefaultConstructor = type.GetConstructors().First();
```

Comment:
Is there a scenario where this will fail? Should I fall back on `LogError`?  I can't think of a situation, except some weird case where another weaver created the non-default constructor which would leave us with not sequence point below, but in that case maybe the general purpose error handler is good enough?